Solution by Ruben Cruz

RHDZMOTA opened the pull request.

Commit:
Ruben Cruz challenge implentation

oscarvarto commented on src/main/scala/com/intersys/mdc/challenge/exercises/problems/Problem1.scala:
```diff
@@ -41,9 +42,15 @@ case object Problem1 extends Problem {
+            var mixed = ""
+            var pos = 0
+
+            for(pos <- 0 to maxStringSize - 1) {
```
👍 Correct solution

oscarvarto commented on src/main/scala/com/intersys/mdc/challenge/exercises/problems/Problem4.scala:
```diff
@@ -45,6 +45,10 @@ case object Problem4 extends Problem {
     */
 
+  implicit class StringOps(string: String) {
+    def asInt: Option[Int] = scala.util.Try(Some(string.toInt)).toOption.getOrElse(None)
```
👎 Not clean.
This makes evident a lack of understanding of `Option` usage patterns.

oscarvarto replied:
This exercise could be used to help Ruben understand for comprehensions and the usage of Option patterns.

oscarvarto commented on src/main/scala/com/intersys/mdc/challenge/exercises/problems/Problem4.scala:
```diff
@@ -55,15 +59,42 @@ case object Problem4 extends Problem {
-          def calculate(ops: String, a: Int, b: Int): Option[Int] = ???
+          def calculate(ops: String, a: Int, b: Int): Option[Int] = {
+            ops match {
+              case "sum" => Some(a+b)
```
👍 Pattern matching

oscarvarto commented on src/main/scala/com/intersys/mdc/challenge/exercises/problems/Problem4.scala:
```diff
@@ -55,15 +59,42 @@ case object Problem4 extends Problem {
+              case "division" => try {
+                Some(a/b)
+              } catch {
+                case _: Exception => None
```
👍 for considering the case when arithmetic exceptions (division by zero) might occur.

oscarvarto commented on src/main/scala/com/intersys/mdc/challenge/exercises/problems/Problem4.scala:
```diff
@@ -55,15 +59,42 @@ case object Problem4 extends Problem {
+            val b = params("b").asInt
+            val operation = params("operation")
+
+            if(a == None || b == None) {
```
👎 Could have used for comprehensions (flatMap and map) for the Option monad. Not idiomatic scala code.

oscarvarto commented on src/main/scala/com/intersys/mdc/challenge/exercises/problems/Problem4.scala:
```diff
@@ -55,15 +59,42 @@ case object Problem4 extends Problem {
+              None
+            }
+            else {
+              val result = calculate(operation, a.get, b.get)
```
👎 Not correct usage of optionals here (`.get`)

emmanuelvallejoIntersys committed:
Update Problem1.scala

Emmanuel Vallejo: Instructions on directives file indicates that the point on this exersice is for using scala collections (zip, zipWith, etc), map, flatMap, or recursion. This solution works but is only using strings and for cycles
